roles/lib_openshift/library/oc_adm_ca_server_cert.py

```diff
@@ -343,7 +344,9 @@ def _write(filename, contents):
         tmp_filename = filename + '.yedit'
 
         with open(tmp_filename, 'w') as yfd:
+            fcntl.flock(yfd, fcntl.LOCK_EX | fcntl.LOCK_NB)
```

Comment:
This will raise 'EWOULDBLOCK' (IOError in python) if the lock cannot be established, causing the module to fail.

We might want to catch and fail with a more user-friendly message.

Reply:
@ashcrow @michaelgugino, We could handle that.  What do you think about implementing a small retry here to attempt to gain the lock after a small short sleep interval?

Reply:
I don't think that's necessary, just remove the fcntl.LOCK_NB and the program will block until exclusive lock is granted.

Reply:
I'm ok with either way as long as overwriting the file with whatever was read at the time of block is alright. Otherwise it could be problematic.

Reply:
Of course, if we want to add a timeout and give user a pretty message about file being locked by another process, we could use LOCK_NB, catch and don't raise until timeout expired, or we could implement a signal handler, or run flock command in a subprocess and use it's timeout semantics.

Reply:
Multiple runs of ansible simultaneously against the same cluster is probably going to not go well overall, IMO.  Probably shouldn't do that.

Reply:
@michaelgugino true